core/src/main/java/hudson/model/User.java

```diff
@@ -731,10 +732,43 @@ public synchronized void save() throws IOException, FormValidation {
             throw FormValidation.error(Messages.User_IllegalFullname(fullName));
         }
         if(BulkChange.contains(this))   return;
-        getConfigFile().write(this);
+        synchronized (saving) {
+            saving.add(this);
+        }
+        try {
+            getConfigFile().write(this);
```

Comment:
Default methods are not an option since the `saving` list cannot be kept in an interface. Could be in a utility class somewhere.